tests/platform_tests/conftest.py

```diff
@@ -144,16 +113,80 @@ def service_time_check(message, status):
         with open(filepath) as json_file:
             report = json.load(json_file)
             service_restart_times.update(report)
-    result = service_restart_times
-    logging.info(json.dumps(result, indent=4))
+
+    return service_restart_times
+
+def analyze_sairedis_rec(messages):
+    state_times = dict()
+    state_patterns = {
+        "Start creating switch": re.compile(r'.*\|c\|SAI_OBJECT_TYPE_SWITCH.*'),
+        "Finish creating switch": re.compile(r'.*\|g\|SAI_OBJECT_TYPE_SWITCH.*SAI_SWITCH_ATTR_DEFAULT_VIRTUAL_ROUTER_ID.*'),
+        "Set default route": re.compile(r'.*\|(S|s)\|SAI_OBJECT_TYPE_ROUTE_ENTRY.*0\.0\.0\.0/0.*SAI_ROUTE_ENTRY_ATTR_PACKET_ACTION=SAI_PACKET_ACTION_FORWARD.*')
+    }
+    states = ["Start creating switch", "Finish creating switch", "Set default route"]
+
+    for message in messages:
+        for state, pattern in state_patterns.items():
+            if re.search(pattern, message):
+                timestamp = message.split("|")[0].strip()
+                state_times[state] = timestamp
+
+    for state in states:
+        logging.info("{:25s} : {}".format(state, state_times[state] if state in state_times else "N/A"))
+
+    return state_times
+
+@pytest.fixture()
+def advanceboot_loganalyzer(duthosts, rand_one_dut_hostname, request):
+    """
+    Advance reboot log analysis.
+    This fixture starts log analysis at the beginning of the test. At the end,
+    the collected expect messages are verified and timing of start/stop is calculated.
+
+    Args:
+        duthosts : List of DUT hosts
+        rand_one_dut_hostname: hostname of a randomly selected DUT
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    loganalyzer = LogAnalyzer(ansible_host=duthost, marker_prefix="test_advanced_reboot", 
+                    additional_files={'/var/log/swss/sairedis.rec': 'recording on: /var/log/swss/sairedis.rec'})
+    marker = loganalyzer.init()
+    loganalyzer.load_common_config()
+
+    ignore_file = os.path.join(TEMPLATES_DIR, "ignore_boot_messages")
+    expect_file = os.path.join(TEMPLATES_DIR, "expect_boot_messages")
+    ignore_reg_exp = loganalyzer.parse_regexp_file(src=ignore_file)
+    expect_reg_exp = loganalyzer.parse_regexp_file(src=expect_file)
+
+    loganalyzer.ignore_regex.extend(ignore_reg_exp)
+    loganalyzer.expect_regex = []
+    loganalyzer.expect_regex.extend(expect_reg_exp)
+    loganalyzer.match_regex = []
+
+    yield
+
+    result = loganalyzer.analyze(marker, fail=False)
+    analyze_result = dict()
+    for key, messages in result["expect_messages"].items():
+        if "syslog" in key:
+            service_restart_times = analyze_syslog(duthost, messages)
+            if service_restart_times is not None:
+                analyze_result["Services"] = service_restart_times
+                logging.info(json.dumps(analyze_result["Services"], indent=4))
```

Comment:
Can we move this console logging out of loop, and log for all of `analyze_result`?

Reply:
Done